### ISSUE TITLE
[Bug]: setWindowOpenHandler of webContents not called while new window open.

### ISSUE THREAD

**Issue by patil-milind** [electron/electron#28656]
### Preflight Checklist

I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a feature request that matches the one I want to file, without success.

### Electron Version

12.0.0

### What operating system are you using?

Windows

### Operating System Version

Windows 10

### What arch are you using?

x64

### Last Known Working Electron version

12.0.0

### Expected Behavior

Electron version: 12.0.0
**Problem:** Need to disable new windows popup,  for this use the  event and its working but this is deprecated in Electron 12.0
```
win.webContents.on('new-window'', event => {
    event.preventDefault()
  })
```
So instead of above used the suggested alternative handler but this is not getting called during new windows open or mouse mid click. 
```
win.webContents.setWindowOpenHandler(() => {
            return { action: 'deny' };
     });
```
**Expected behavior:**  On mouse mid click new windows creation should prevented.  

### Actual Behavior

**Actual** : New window opens on mouse windows click.

**Comment by VerteDinde** [comment]
Hey @patil-milind, thanks for reporting this issue! Correct me if I'm wrong, but I think you may be experiencing the same issue as this user: https://github.com/electron/electron/issues/27967 We have a fix for this merged into master, and will backport it into 12-x-y - the fix is already available in the newest 13 beta. I'll close this issue once it's backported into 12.

**Comment by patil-milind** [comment]
@VerteDinde  Yes its same issue as #27967 .

**Comment by nornagon** [comment]
Duplicate of #27967

**Comment by VerteDinde** [comment]
@patil-milind This fix is available in 12.0.5 🙂 Let us know if it still doesn't work for you!

**Comment by Delagen** [comment]
@VerteDinde I catch download window with this, and calling `{action:'deny'}` freeze main frame

**Comment by nvdai2401** [comment]
> catch download window with this, and calling `{action:'deny'}` freeze main frame

I'm dealing with the same problem as you

**Comment by andresavic** [comment]
I get the error message 
` this.win.webContents.setWindowOpenHandler(({ url }) => {
            return { action: 'deny' }
        })`

> The window open handler response must be an object, but was instead of type 'undefined'.

and the main window freezes as well.

**Comment by nornagon** [comment]
@andresavic @nvdai2401 @Delagen that sounds like #29509, which is recently fixed by #29851.
